src/groups/metric.jl

```diff
@@ -0,0 +1,132 @@
+@doc doc"""
+    has_invariant_metric(G::AbstractGroupManifold, conv::ActionDirection) -> Val
+
+Return `Val(true)` if the metric on the group $G$ is invariant under translations by the
+specified direction, that is, given a group $G$, a left- or right group translation map $τ$,
+and a metric $g_e$ on the Lie algebra, a $τ$-invariant metric at any point $x ∈ G$ is
+defined as
+
+````math
+g_x(v, w) = g_{τ_y x}((\mathrm{d}τ_y)_x v, (\mathrm{d}τ_y)_x w),
+````
+
+for $v,w ∈ T_x G$ and all $y ∈ G$, where $(\mathrm{d}τ_y)_x$ is the action of the
+differential of translation by $y$ evaluated at $x$ (see [`translate_diff`](@ref)).
+"""
+function has_invariant_metric(M::Manifold, conv::ActionDirection)
+    return has_invariant_metric(M, conv, is_decorator_manifold(M))
+end
+function has_invariant_metric(M::Manifold, conv::ActionDirection, ::Val{true})
+    return has_invariant_metric(M.manifold, conv)
+end
+has_invariant_metric(::Manifold, ::ActionDirection, ::Val{false}) = Val(false)
+
+"""
+    has_biinvariant_metric(G::AbstractGroupManifold) -> Val
+
+Return `Val(true)` if the metric on the manifold is bi-invariant, that is, if the metric
+is both left- and right-invariant (see [`has_invariant_metric`](@ref)).
+"""
+function has_biinvariant_metric(M::Manifold)
+    return Val(
+        (
```

Comment:
Why do you need these additional round brackets? It looks like to would return `Val((true,))` or `Val((false,))` instead of `Val(true)` or `Val(false)`.

Reply:
> Why do you need these additional round brackets?

Unsuccessful attempt to avoid dangling `&&` sign. Will remove. The comma is outside the inner parenthesis, so the two expressions are equivalent.